CadRevealFbxProvider.Tests/Attributes/ScaffoldingCsvLineParserTests.cs

```diff
@@ -0,0 +1,184 @@
+    [Test]
+    public void ExtractSingleWeightFromCsvRow_WithValidWeightHeader_ReturnsWeight()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Weight kg", "Other" }, Values = new[] { "42", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractSingleWeightFromCsvRow(row);
+        Assert.That(result, Is.EqualTo("42"));
+    }
+
+    [Test]
+    public void ExtractSingleWeightFromCsvRow_WithEmptyWeightHeader_ReturnsNull()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Weight kg", "Other" }, Values = new[] { "", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractSingleWeightFromCsvRow(row);
+        Assert.That(result, Is.Null);
+    }
+
+    [Test]
+    public void ExtractSingleDescriptionFromCsvRow_WithManufacturerFalse_ReturnsDescriptionWithManufacturer()
+    {
+        var row = new TestCsvLine { Headers = new[] { "HAKI Description", "Other" }, Values = new[] { "desc", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractSingleDescriptionFromCsvRow(row, false);
+        Assert.That(result, Does.Contain("desc"));
+    }
+
+    [Test]
+    public void ExtractKeyFromCsvRow_WithValidKeyHeader_ReturnsKey()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Item code", "Other" }, Values = new[] { "KEY123", "foo" } };
+        var result = ScaffoldingCsvLineParser.ExtractKeyFromCsvRow(row, 0, "Item code");
+        Assert.That(result, Is.EqualTo("KEY123"));
+    }
+
+    [Test]
+    public void ExtractKeyFromCsvRow_WithMissingKeyHeader_ThrowsException()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Item code", "Other" }, Values = new[] { "", "foo" } };
+        Assert.That(
+            () => ScaffoldingCsvLineParser.ExtractKeyFromCsvRow(row, 0, "Item code"),
+            Throws.TypeOf<ScaffoldingAttributeParsingException>()
+        );
+    }
+
+    [Test]
+    public void IsNumericSapColumn_WithNumericHeader_ReturnsTrue()
+    {
+        var row = new TestCsvLine { Headers = new[] { "Work order", "Other" }, Values = new[] { "123", "foo" } };
+        var numericHeaders = new List<string> { "Work order" };
+        var result = ScaffoldingCsvLineParser.IsNumericSapColumn(row, 0, numericHeaders);
+        Assert.That(result, Is.True);
+    }
+
+    [Test]
+    public void IsNumericSapColumn_WithNonNumericHeader_ReturnsFalse()
+    {
+        var row = new TestCsvLine { Headers = new[] { "NotNumeric", "Other" }, Values = new[] { "123", "foo" } };
+        var numericHeaders = new List<string> { "Work order" };
+        var result = ScaffoldingCsvLineParser.IsNumericSapColumn(row, 0, numericHeaders);
+        Assert.That(result, Is.False);
+    }
+
+    [Test]
+    public void ExtractColumnValueFromRow_WithMultipleNonEmptyValues_ThrowsException()
+    {
+        var row = new TestCsvLine { Headers = new[] { "B", "B" }, Values = new[] { "2", "3" } };
+        Assert.That(() => ScaffoldingCsvLineParser.ExtractColumnValueFromRow("B", row), Throws.Exception);
```

Comment:
I think this should not throw just a generic exception, but then there is another PR in our stack that can address it ;)

Reply:
Yeah, lets consider solidifying in your PR.